### PR TITLE
fix: disable cache for injected file.

### PR DESCRIPTION
**Pull request by davidfestal**
## Description

Backend merged config is injected by the `app-backend` upstream plugin inside one of the Javascript
static assets before service static assets.
So we should never allow this specific JS file to be cached on the client side,
so that any change in the configuration can be seen on the frontend
side after a backend restart.
That's particularly critical to get the latest UI configuration
of added dynamic frontend plugins.

## Which issue(s) does this PR fix

- Fixes No issue number.

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

**Review by tumido — APPROVED** [review]
/lgtm
/approve

File: packages/backend/src/plugins/app.ts [hunk 1]
```diff
@@ -1,14 +1,65 @@
...
+  const jsFiles = files.filter(file => file.endsWith('.js'));
+  let injectedJSFile: string | undefined = undefined;
+
+  for (const jsFile of jsFiles) {
+    const path = resolvePath(staticDir, jsFile);
+
+    const content = await fs.readFile(path, 'utf8');
+    if (content.includes('__APP_INJECTED_')) {
+      injectedJSFile = jsFile;
+      break;
```
Comment (Hyperkid123) [inline packages/backend/src/plugins/app.ts:27]: Is it 100% confirmed that there will only ever be one file with the mark? I don't think its that much extra work to assume there can be more than one file that gest mutated at runtime.

File: packages/backend/src/plugins/app.ts [hunk 2]
```diff
@@ -1,14 +1,65 @@
...
   });
+
+  const enclosingRouter = Router();
+  if (injectedJSFile) {
+    env.logger.info(
+      `Setting up static router for injected Javascript file ${injectedJSFile}`,
+    );
+
+    enclosingRouter.get(
+      `/static/${injectedJSFile}`,
```
Comment (Hyperkid123) [inline packages/backend/src/plugins/app.ts:45]: We can iterate over the array to set up multiple routes with this rule

File: packages/backend/src/plugins/app.ts (hunk 1 above)
Comment (davidfestal) [reply packages/backend/src/plugins/app.ts:27]: > Is it 100% confirmed that there will only ever be one file with the mark

It is how the `backend-app` plugins works. See the `return` lines in https://github.com/backstage/backstage/blob/master/plugins/app-backend/src/lib/config.ts#L55 and https://github.com/backstage/backstage/blob/master/plugins/app-backend/src/lib/config.ts#L64

File: packages/backend/src/plugins/app.ts (hunk 2 above)
Comment (davidfestal) [reply packages/backend/src/plugins/app.ts:45]: Is it really useful based on my answer [above](https://github.com/janus-idp/backstage-showcase/pull/710#discussion_r1383255152) ?

File: packages/backend/src/plugins/app.ts (hunk 1 above)
Comment (Hyperkid123) [reply packages/backend/src/plugins/app.ts:27]: OK. Well. Then we are good.

File: packages/backend/src/plugins/app.ts (hunk 2 above)
Comment (Hyperkid123) [reply packages/backend/src/plugins/app.ts:45]: Nope

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**